script/arrays.js

```diff
@@ -0,0 +1,221 @@
+// minimum
+
+
+// 1. Створи масив «Список покупок». Кожен елемент масиву є об'єктом, який містить назву продукту, 
+// кількість і куплений він чи ні, ціну за одиницю товару, сума. Написати кілька функцій для роботи з таким масивом:
+
+
+function Product(name, quantity, bought, unitPrice) {
+    this.name = name;
+    this.quantity = quantity;
+    this.bought = bought;
+    this.unitPrice = unitPrice;
+    let sum = quantity * unitPrice;
+    this.sum = sum;
+}
+
+let shopingList = new Array();
+shopingList[0] = new Product("apple", 2, false, 5);
+shopingList[1] = new Product("milk", 1, true, 30);
+shopingList[2] = new Product("orange", 5, false, 10);
+shopingList[3] = new Product("sausage", 1, false, 100);
+shopingList[4] = new Product("bread", 1, false, 27);
+shopingList[5] = new Product("lemon", 2, true, 7);
+shopingList[6] = new Product("avocado", 1, false, 55);
+
+//  Виводити весь список на екран таким чином, щоб спочатку йшли продукти, що ще не придбані, а потім - ті, що вже придбали.
+
+shopListOrdered = (arr) => {
+    console.log("Shopping List");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == true) {
+            continue;
+        }
+        console.log(arr[i]);
+    }
+    for (let i = 0; i < arr.length; i++) {
```

Comment:
Duplicate `for` loops. It's a good place to refactor and use another strategy. 2 loops will take x2 more time for given calculations. Imagine we have a complex task running inside these loops... 😢